src/ConnectionAutomaton.cpp

```diff
@@ -320,13 +320,13 @@ void ConnectionAutomaton::onConnectionFrame(
   switch (type) {
     case FrameType::KEEPALIVE: {
       Frame_KEEPALIVE frame;
-      if (!deserializeFrameOrError(frame, std::move(payload))) {
+      if (!deserializeFrameOrError(isResumable_, frame, std::move(payload))) {
```

Comment:
I understand why the hack is necessary, but I think the implementation in deserialize method is not the desired one.
Assume this. Mobile clients are always sending RESUMABLE == true. And on Server we have a switch to turn on/off the resumability as a feature. So if client is ON and server is OFF the desired behavior is that we are still talking to each other, but without resumability. With you hack we will not agree on the format of the KEEPALIVE frame.
I would suggest a hack which would check the length of the IOBuf in the deserialize method and if it is sizeof(frame-in-spec)+sizeof(position), then we are getting the resumable position.